programs/bpf_loader/src/lib.rs

```diff
@@ -314,33 +244,31 @@ pub fn create_executor_from_account(
             .get_data()
             .get(programdata_offset..)
             .ok_or(InstructionError::AccountDataTooSmall)?,
+        program.get_owner(),
+        program.get_data().len().saturating_add(programdata_size),
+        deployment_slot,
         use_jit,
         false, /* reject_deployment_of_broken_elfs */
     )?;
-    if let Some(mut tx_executor_cache) = tx_executor_cache {
-        tx_executor_cache.set(
-            *program.get_key(),
-            executor.clone(),
-            false,
-            feature_set.is_active(&delay_visibility_of_program_deployment::id()),
-        );
-    }
-    Ok((executor, Some(create_executor_metrics)))
+    Ok((loaded_program, Some(create_executor_metrics)))
```

Comment:
`tx_executor_cache.set()` is still necessary for programs which are just executed, not being re-/deployed as both code paths go through this function.